packages/web/src/components/tracks-table/TracksTable.tsx

```diff
@@ -153,6 +161,11 @@ export const TracksTable = ({
       const active = index === playingIndex
       const track = cellInfo.row.original
       const isTrackUnlisted = track.is_unlisted
+      const isTrackPremium = isContentUSDCPurchaseGated(track.stream_conditions)
+      const { isFetchingNFTAccess, hasStreamAccess } = trackAccessMap[
+        track.track_id
+      ] ?? { isFetchingNFTAccess: false, hasStreamAccess: true }
+      const isLocked = !isFetchingNFTAccess && !hasStreamAccess
```

Comment:
one general thing that I considered was refactoring these things here. Seems like a lot of these "isTrack XYZ" are repeated in these `renderXYZ` methods. For now I stuck with the same pattern

Reply:
yeah this does feel like we could do a hook that gets you the value you need, or a helper that returns a `ContentType` enum value

Reply:
yea I'm gonna leave it for the moment. It's just a mild nuisance anyways